indra/newview/llvoicevivox.cpp

```diff
@@ -6437,37 +6437,52 @@ void LLVivoxVoiceClient::notifyStatusObservers(LLVoiceClientStatusObserver::ESta
 			}
 		}
 	}
-		
+	
 	LL_DEBUGS("Voice") 
 		<< " " << LLVoiceClientStatusObserver::status2string(status)  
 		<< ", session URI " << getAudioSessionURI() 
 		<< ", proximal is " << inSpatialChannel()
         << LL_ENDL;
 
-	for (status_observer_set_t::iterator it = mStatusObservers.begin();
-		it != mStatusObservers.end();
-		)
-	{
-		LLVoiceClientStatusObserver* observer = *it;
-		observer->onChange(status, getAudioSessionURI(), inSpatialChannel());
-		// In case onError() deleted an entry.
-		it = mStatusObservers.upper_bound(observer);
-	}
+    // this function is called from a coroutine, shuttle application hook back to main loop
+    auto mainqueue = LL::WorkQueue::getInstance("mainloop");
```

Comment:
I think I'd prefer to see this `getInstance()` call closer to line 6478 where you test it.